Feature/extend holdings endpoint by performance with currency effect for cash

KenTandrian opened the pull request:
Hi @dtslvr, this PR resolves #5612. Please take a look :)

### Changes
* Extended holdings endpoint by performance with currency effect for cash:
  * The `getDetails` method in `PortfolioService` now fetches historical exchange rate data for all currencies associated with the user's cash accounts. This is done once per call to ensure efficiency.
  * The `getCashPositions` method has been updated to accept this historical data. It then calculates the `netPerformanceWithCurrencyEffect` for each cash position by comparing the current value of the cash in the base currency against its value at the time the account was created.
  * The `netPerformancePercentWithCurrencyEffect` is then derived from `netPerformanceWithCurrencyEffect` value relative to the initial investment.
* Updated changelog.
* Added comments to clarify the new logic.

KenTandrian commented:
I realized that `this.calculatorFactory.createCalculator()` is being called in many places, while I am only handling one of them. Therefore, the cash synthetic orders calculation is moved to `OrderService` so that it takes effect to all of them.

KenTandrian commented:
hi @dtslvr, sorry for the delayed finalization of this PR.
The PR is ready to review. You can also take a look at the changes in the PR description. Thanks!

Anyway, I think we should work on adding JSDoc documentations on some important methods so that contributors (and also potentially AI agents) can understand better.

dtslvr commented on apps/api/src/app/order/order.service.ts:
```diff
@@ -317,6 +323,108 @@ export class OrderService {
+
+    for (const account of cashDetails.accounts) {
+      const { balances } = await this.accountBalanceService.getAccountBalances({
+        filters: [{ id: account.id, type: 'ACCOUNT' }],
```
Move down ⬇️

dtslvr commented on apps/api/src/app/portfolio/calculator/roai/portfolio-calculator.ts:
```diff
@@ -294,6 +298,7 @@ export class RoaiPortfolioCalculator extends PortfolioCalculator {
       itemType: 'start',
       quantity: new Big(0),
       SymbolProfile: {
+        assetSubClass: isCash ? 'CASH' : undefined,
```
Move down ⬇️

dtslvr commented on apps/api/src/app/portfolio/calculator/roai/portfolio-calculator.ts:
```diff
@@ -307,6 +312,7 @@ export class RoaiPortfolioCalculator extends PortfolioCalculator {
       feeInBaseCurrency: new Big(0),
       itemType: 'end',
       SymbolProfile: {
+        assetSubClass: isCash ? 'CASH' : undefined,
```
Move down ⬇️

KenTandrian commented on apps/api/src/app/portfolio/calculator/portfolio-calculator.ts:
```diff
@@ -136,12 +136,15 @@ export abstract class PortfolioCalculator {
             date: format(date, DATE_FORMAT),
             fee: new Big(feeInAssetProfileCurrency),
             quantity: new Big(quantity),
+            SymbolProfile: {
```
It's actually not needed. Thanks for pointing that out!

dtslvr commented on apps/api/src/app/portfolio/calculator/roai/portfolio-calculator-cash-performance.spec.ts:
```diff
@@ -0,0 +1,254 @@
+  let redisCacheService: RedisCacheService;
+
+  beforeAll(() => {
+    exportResponse = loadExportFile(
```
This is not really in use as data is hard-coded below...

dtslvr commented on apps/api/src/app/portfolio/calculator/roai/portfolio-calculator-cash-performance.spec.ts:
```diff
@@ -0,0 +1,254 @@
+            assetSubClass: 'CASH',
+            currency: 'USD',
+            dataSource: 'MANUAL',
+            name: 'US Dollar',
```
`name: 'USD'`

dtslvr commented on apps/api/src/app/portfolio/calculator/roai/portfolio-calculator-cash-performance.spec.ts:
```diff
@@ -0,0 +1,254 @@
+      jest.spyOn(currentRateService, 'getValues').mockResolvedValue({
+        dataProviderInfos: [],
+        values: [],
+        errors: []
```
Move up ⬆️

KenTandrian commented:
@dtslvr I have made some changes to show the names for currencies in the holdings table.

### Problem
Cash assets (e.g., USD, CHF) were appearing with blank names in the holdings list. This occurred because `PortfolioService.getDetails` relies on `assetProfile.name`, which is `undefined` for currency positions.

This happened because `assetProfile` depends on `symbolProfiles`, which does not include cash.

https://github.com/ghostfolio/ghostfolio/blob/95cbd01a4c5becac550c69a69f74fa35fcf5502d/apps/api/src/app/portfolio/portfolio.service.ts#L556-L558

### Solution
This PR implements a fallback naming strategy for cash assets:
1.  **Updated Model**: Added `assetSubClass` to the [TimelinePosition](https://github.com/ghostfolio/ghostfolio/blob/main/libs/common/src/lib/models/timeline-position.ts) interface to distinguish asset types.
2.  **Logic Update**: Modified [PortfolioCalculator](https://github.com/ghostfolio/ghostfolio/blob/main/apps/api/src/app/portfolio/calculator/portfolio-calculator.ts) to propagate `assetSubClass` to position objects.
3.  **Fallback Name**: In [PortfolioService](https://github.com/ghostfolio/ghostfolio/blob/main/apps/api/src/app/portfolio/portfolio.service.ts), if an asset is identified as `CASH`, the name defaults to `${symbol} (Cash)` (e.g., "USD (Cash)") instead of remaining blank.

dtslvr reviewed (approved):
Huge thanks for your persistence on this one, @KenTandrian! 🙌🏻
Really appreciate the hard work to get it merged.
